internal/services/oidc.go

```diff
@@ -2,39 +2,48 @@ package services
 
 import (
 	"context"
-	"crypto/rand"
 	"encoding/json"
 	"errors"
-	"io"
+	"fmt"
 	"time"
 
 	"github.com/coreos/go-oidc/v3/oidc"
 	"github.com/golang-jwt/jwt/v4"
+	"golang.org/x/oauth2"
+
 	"github.com/ydataai/authentication-service/internal/clients"
+	"github.com/ydataai/authentication-service/internal/configurations"
+	authErrors "github.com/ydataai/authentication-service/internal/errors"
 	"github.com/ydataai/authentication-service/internal/models"
 	"github.com/ydataai/authentication-service/internal/storages"
 	"github.com/ydataai/go-core/pkg/common/logging"
-	"golang.org/x/oauth2"
 )
 
 // OIDCService defines the oidc server struct.
 type OIDCService struct {
+	configuration  configurations.OIDCServiceConfiguration
 	client         clients.OIDCClient
 	sessionStorage *storages.SessionStorage
 	logger         logging.Logger
 }
 
 // NewOIDCService creates a new OIDC Service.
 func NewOIDCService(logger logging.Logger,
+	configuration configurations.OIDCServiceConfiguration,
 	client clients.OIDCClient,
-	sessionStorage *storages.SessionStorage) *OIDCService {
-	return &OIDCService{
+	sessionStorage *storages.SessionStorage) OIDCService {
+	return OIDCService{
+		configuration:  configuration,
 		client:         client,
 		sessionStorage: sessionStorage,
 		logger:         logger,
 	}
 }
 
+var (
+	hmacSecret = []byte("YData") // For HMAC signing method, the key can be any []byte
```

Comment:
🤔  doesn't seem to me a very secure secret 😅 

let's inject this, btw and this will be generated with a random generator

Reply:
thanks for the explanation.